Test/Integration/Product/ReplicaIndexingTest.php

```diff
@@ -0,0 +1,333 @@
+        $sorting[] = [
+            'attribute'      => $sortAttr,
+            'sort'           => $sortDir,
+            'sortLabel'      => $sortAttr,
+            'virtualReplica' => 1
+        ];
+        $this->configHelper->setSorting($sorting);
+
+        $this->assertConfigInDb(ConfigHelper::SORTING_INDICES, json_encode($sorting));
+
+        $this->refreshConfigFromDb();
+
+        $this->assertSortingAttribute($sortAttr, $sortDir);
+
+        // Cannot use config fixture because we have disabled db isolation
+        $this->setConfig(ConfigHelper::IS_INSTANT_ENABLED, 1);
+
+        $this->indicesConfigurator->saveConfigurationToAlgolia(1);
+        $this->algoliaHelper->waitLastTask();
+
+        // Assert replica config created
+        $currentSettings = $this->algoliaHelper->getSettings($indexName);
+        $this->assertArrayHasKey('replicas', $currentSettings);
+
+        $sortIndexName = $indexName . '_' . $sortAttr . '_' . $sortDir;
+
+        $this->assertTrue($this->isVirtualReplica($currentSettings['replicas'], $sortIndexName));
+        $this->assertFalse($this->isStandardReplica($currentSettings['replicas'], $sortIndexName));
+
+        // Assert replica index created
+        $replicaSettings = $this->assertReplicaIndexExists($indexName, $sortIndexName);
+        $this->assertVirtualReplicaRanking($replicaSettings, "$sortDir($sortAttr)");
+
+        // Restore prior state (for this test only)
+        $this->configHelper->setSorting($ogSortingState);
+        $this->setConfig(ConfigHelper::IS_INSTANT_ENABLED, 0);
+    }
+
+    /**
+     * ConfigHelper::setSorting uses WriterInterface which does not update unless DB isolation is disabled
+     * This provides a workaround to test using MutableScopeConfigInterface with DB isolation enabled
+     */
+    protected function mockSortUpdate(string $sortAttr, string $sortDir, array $attr): void
+    {
+        $sorting = $this->configHelper->getSorting();
+        $existing = array_filter($sorting, function ($item) use ($sortAttr, $sortDir) {
+           return $item['attribute'] === $sortAttr && $item['sort'] === $sortDir;
+        });
+
+
+        if ($existing) {
+            $idx = array_key_first($existing);
+            $sorting[$idx] = array_merge($existing[$idx], $attr);
+        }
+        else {
+            $sorting[] = array_merge(
+                [
+                    'attribute' => $sortAttr,
+                    'sort'       => $sortDir,
+                    'sortLabel'  => $sortAttr
+                ],
+                $attr
+            );
+        }
+        $this->setConfig(ConfigHelper::SORTING_INDICES, json_encode($sorting));
+    }
+
+    /**
+     * @depends testReplicaSync
+     * @magentoConfigFixture current_store algoliasearch_instant/instant/is_instant_enabled 1
+     * @throws AlgoliaException
+     * @throws ExceededRetriesException
+     * @throws \ReflectionException
+     */
+    public function testReplicaRebuild(): void
```

Comment:
Curious again, why is this mock and reflection implementation needed here ? Is it only for protected methods ?

Reply:
Yup just for protected methods - I wanted to test the command class in its entirety but `execute` is protected. I also needed to mock the Input/OutputInterface.